analyzer/src/main/kotlin/managers/Yarn2.kt

```diff
@@ -126,21 +192,530 @@ class Yarn2(
+     * The list of dependencies of the constructed object is returned.
+     */
+    private fun parsePackage(
+        json: ObjectNode,
+        definitionFile: File
+    ): Map<DependencyType, Pair<Identifier, List<Identifier>>> {
+        val value = json["value"].textValue()
+
+        val header = requireNotNull(packagesHeaders[value]) {
+            "No package header found for '$value'."
+        }
+
+        val (namespace, name) = Npm.splitNamespaceAndName(header.rawName)
+        val childrenNode = json["children"]
+        val version = childrenNode["Version"].textValue()
+
+        val manifest = childrenNode["Manifest"] as ObjectNode
+        // Yarn's manifests contain licenses in an element with an uppercase 'L'. To leverage existing license parsing
+        // code, an extra property with lowercase 'L' is added.
+        manifest["License"].takeUnless {
+            it is NullNode
+        }?.also {
+            manifest.set<ObjectNode>("license", it)
+        }
+        val declaredLicenses = Npm.parseLicenses(manifest)
+
+        val cache = childrenNode["Cache"]
+
+        var homepageUrl = manifest["Homepage"].textValueOrEmpty()
+
+        var hash = Hash.create(cache["Checksum"].textValueOrEmpty())
+
+        val id = if (header.type == "workspace") {
+            val projectFile = definitionFile.parentFile.resolve(header.version).resolve(definitionFile.name)
+            val workingDir = definitionFile.parentFile
+
+            val additionalData = getProjectAdditionalData(workingDir, name, version)
+            val id = Identifier("Yarn2", namespace, name, version)
+            allProjects += id to Project(
+                id = id.copy(type = managerName),
+                definitionFilePath = VersionControlSystem.getPathInfo(projectFile).path,
+                declaredLicenses = declaredLicenses,
+                vcs = additionalData.vcsFromPackage,
+                vcsProcessed = processProjectVcs(definitionFile.parentFile, additionalData.vcsFromPackage, homepageUrl),
+                homepageUrl = homepageUrl
+            )
+            id
+        } else {
+            val versionFromLocator = cleanYarn2VersionString(header.version)
+            val details: AdditionalData = requireNotNull(packagesDetails["${header.rawName}@$versionFromLocator"]) {
+                "No package details found for '${header.rawName}' at version '$versionFromLocator'."
+            }
+
+            @Suppress("MaxLineLength")
+            // TODO : It is unclear how hashes coming from Yarn are created. They are present in the .yarn.lock file
+            //  under the name "checksum" and they are 130 characters long. It seems these are SHA512 hashes with two
+            //  extra characters. For instance :
+            //  8/b296c92c4b969e973260e47523207cd5769abd27c245a68c26dc7a0fe8053c55bb04360237cb51cab1df52be939da77150ace99ad331fb7fb13b3423ed73ff3d
```

Comment:
I'm not seeing those extra 2 chars in the committed Yarn2 `yarn.lock` files...

Reply:
I think it happens only if you use NPM package with SHA512 hashes.
None of the one in the test data packages has such hash....

Reply:
I checked again: you get such hash only in the output of `yarn info` e.g.:
```
{
    "value": "@types/node@npm:12.6.8",
    "children": {
        "Version": "12.6.8",
        "Manifest": {
            "License": "MIT",
            "Homepage": null
        },
        "Cache": {
            "Checksum": "8/b0079ab6c10c6f62e9f6dc75b15446a0c21e476b9ff23977639a96407c34fb1354fbfa8edd6f18cc80b1f21e21d393eefbf5a2d48034f74c4c770d4ce2608d07",
            "Path": "analyzer/src/funTest/assets/projects/synthetic/yarn2/.yarn/cache/@types-node-npm-12.6.8-edf14989b6-b0079ab6c1.zip",
            "Size": 103602
        },
        "Dependents": [
            "parse5@npm:3.0.3"
        ]
    }
}
```
`Yarn.lock` indeed contains only the hash. This `8/` is a mystery to me.

Reply:
I rephrased the TODO.

Reply:
I adapted the code.

BTW after looking at your pointer I found this:
https://github.com/yarnpkg/berry/blob/1a489d1aed51a447022277b22695f783b4065e29/packages/yarnpkg-core/sources/Project.ts#L354

Could it be that this `8/` is the cache key ?

Reply:
I am starting to wonder if this cache field of `yarn info` is not the checksum of the package but the checksum of the Yarn cache entry.
Therefore, the correct hash would be the one from `yarn npm info` (the `details` in code).
What do you think ?

Reply:
> Could it be that this `8/` is the cache key ?

Looks like so, very weird.

> Therefore, the correct hash would be the one from `yarn npm info` (the `details` in code).

Well, that's what I mean with "will the remaining string be the correct SHA512 for the package". If the hashes from `yarn npm info` and the hash with stripped "8/" do not match (although being of the same length), I'd only use the hash from `yarn npm info` to be on the safe side.

Reply:
Changed to use the hashes from `yarn npm info`.
Also removed fetching the yarn cache information altogether.